syntax = "proto3";

package proxy.filters.access_log.escape_filter.v2;

option java_package = "proxy.filters.access_log.escape_filter.v2;";
option java_outer_classname = "EscapeFilterProto";
option java_multiple_files = true;

import "validate/validate.proto";

message EscapeConfig {
  oneof to_escape {
    option (validate.required) = true;

    string rqx_header = 1 [(validate.rules).string.min_bytes = 1];
    string rpx_header = 2 [(validate.rules).string.min_bytes = 1];
  }
}

message ProtoConfig {
  repeated EscapeConfig to_escapes = 1;
}